pkg/prom/ha/http.go

```diff
@@ -34,10 +36,16 @@ type APIHandler func(r *http.Request) (interface{}, error)
 // WireAPI injects routes into the provided mux router for the config
 // management API.
 func (s *Server) WireAPI(r *mux.Router) {
-	listConfig := s.WrapHandler(s.ListConfigurations)
-	getConfig := s.WrapHandler(s.GetConfiguration)
-	putConfig := s.WrapHandler(s.PutConfiguration)
-	deleteConfig := s.WrapHandler(s.DeleteConfiguration)
+	// PutConfiguration below is wrapped in a nonConcurrentHandler to prevent a
+	// race condition of two conflicting instance configs being applied at the same time.
+	// Since applying config performs KV store-wide validations, it is possible that
+	// concurrently applied configs that conflict each other may not be rejected if
+	// neither conflicting config has been persisted to the KV store yet.
```

Comment:
nit: this comment is bit hard to understand, we can reword it make it easy.

one way to reword it might be :point_down:, but feel free to reword in other ways as well.
> PutConfiguration persists configs to KV store with a Read-Modify-Write (RMW) operation. We take a lock (do not allow concurrent operations) to ensure that config we are reading, modifying and writing is not modified after we read.



ps: this is a nitpick, feel free to ignore it :)